services/api/service.go

```diff
@@ -835,6 +915,55 @@ func (api *RelayAPI) handleGetPayload(w http.ResponseWriter, req *http.Request)
 		if err != nil {
 			log.WithError(err).Error("failed to increment builder-stats after getPayload")
 		}
+
+		builderStatus, err := api.redis.GetBlockBuilderStatus(bidTrace.BuilderPubkey.String())
+		if err != nil {
+			log.WithError(err).Error("failed to get builder status")
+		}
+
+		// Check if the block was valid in the optimistic case.
+		if builderStatus == common.Optimistic {
+			// Set to high-prio while we process the winning block.
+			err := api.datastore.SetBlockBuilderStatus(bidTrace.BuilderPubkey.String(), common.HighPrio)
```

Comment:
this makes a lot of sense! i think in terms of implementing this i would just go for 3 statuses rather than try to encode the semantics of "sub-status", but having `optimisticActive, optimisticDemoted, optimisticGetPayload` in addition to `lowPrio, highPrio, blacklisted` feels correct. 

also agree that a single datastore function that sets the status over the set of builders with matching collateral ids makes sense and will be useful every time we need to modify that collection of statuses! 

i will work on this ASAP. leaving the thread open to track and will circle back!

Reply:
Just going for 3 statuses makes sense to me!

Reply:
I wonder if there's a better name than `optimisticGetPayload`. Maybe `optimisticLocked`?

Reply:
check out https://github.com/michaelneuder/mev-boost-relay/pull/2/commits/51edd33f95b06cffea4bca45b4a3a50623c5f591! this is taking into account the discussion above. breaking down the following cases, i think this covers the expected behavior. 

Case 1. 
1. `optimisticBlockProcesor` receives the block first. Block simulation fails. Builder status set to `common.OptimisticDemoted`. 
2.  `getPayload` sees `common.OptimisticDemoted`, so just inserts the `signedBeaconBlock` into the builder demotion table (doesn't resimulate).

Case 2.
1. `getPayload` receives the block first. Block simulation fails. Builder status set to `common.OptimisticDemoted` and `signedBeaconBlock` inserted into the demotion table.
2. `optimisticBlockProcesor` sees that that status of the builder is no longer `common.OptimisticActive`,  so just skips the simulation.

Case 3. 
1.  `getPayload` receives the block first. Block simulation in progress. Builder status set to `common.OptimisticLocked`.
2. `optimisticBlockProcesor` sees that that status of the builder is no longer `common.OptimisticActive`,  so just skips the simulation.

There are more cases than this, and especially if the block simulation succeeds, we will probably end up simulating the blocks twice. e.g., `optimisticBlockProcesor` gets block first, successfully simulates it so no status change is made. `getPayload` gets the block second and sees status is still `OptimisticActive`, so simulates as well. But I think trying to handle each of these individually adds complexity without really solving much (especially because `getPayload` is a once per slot operaion. 

LMK what you think!